Annotato/Annotato/ViewModels/Annotation/AnnotationMarkdownViewModel.swift

```diff
@@ -0,0 +1,54 @@
+import Foundation
+import CoreGraphics
+import Combine
+
+class AnnotationMarkdownViewModel: AnnotationPartViewModel, ObservableObject {
+    weak var parentViewModel: AnnotationViewModel?
+
+    private(set) var id: UUID
+    private(set) var content: String
+    private(set) var origin: CGPoint
+    private(set) var width: Double
+    internal var height: Double
+
+    @Published private(set) var isEditing = false
+    @Published private(set) var isRemoved = false
+    @Published internal var isSelected = false
+    var isEmpty: Bool {
+        content.isEmpty
+    }
+
+    init(id: UUID, content: String, width: Double, height: Double, origin: CGPoint = .zero) {
+        self.id = id
+        self.content = content
+        self.origin = origin
+        self.width = width
+        self.height = height
+    }
+
+    func toView() -> AnnotationPartView {
+        AnnotationMarkdownView(viewModel: self)
+    }
+
+    func enterEditMode() {
+        isEditing = true
+    }
+
+    func enterViewMode() {
+        isEditing = false
+    }
+
+    func setContent(to newContent: String) {
+        self.content = newContent
+    }
+
+    func remove() {
+        isRemoved = true
+    }
+}
+
+extension AnnotationMarkdownViewModel {
+    var editFrame: CGRect {
```

Comment:
Edit frame is specific to Markdown, because markdown has different frames based on whether it's in edit or view mode, but I may look into generalising this and if so, it can be in AnnotationPartViewModel.